### DISCUSSION TITLE
Installing strongswan on Win10 with the flag "--enable-kernel-libipsec"

### DISCUSSION THREAD

**Discussion by wyw900807** [strongswan/strongswan#1180] [Q&A]
I try to install strongswan 5.9.0 on Win10 with the flag "--enable-kernel-libipsec", the command is as follows:
./configure --disable-defaults --enable-monolithic --enable-static --enable-svc --enable-ikev2 --enable-ikev1 --enable-openssl --enable-nonce --enable-pem --enable-pkcs1 --enable-x509  --enable-socket-win --enable-kernel-wfp --enable-kernel-iph --enable-kernel-libipsec --enable-pubkey --enable-eap-identity --enable-eap-mschapv2 --enable-pki --enable-swanctl --with-swanctldir=swanctl --with-strongswan-conf=strongswan.conf 
But during the compilling, there are some errors:

kernel_libipsec_router.c:314:28: error: 'F_GETFL' undeclared (first use in this function)
  int flags = fcntl(socket, F_GETFL);
                            ^~~~~~~
kernel_libipsec_router.c:314:28: note: each undeclared identifier is reported only once for each function it appears in
kernel_libipsec_router.c:315:38: error: 'F_SETFL' undeclared (first use in this function); did you mean 'FD_SET'?
  return flags != -1 && fcntl(socket, F_SETFL, flags | O_NONBLOCK) != -1;
                                      ^~~~~~~
                                      FD_SET
kernel_libipsec_router.c:315:55: error: 'O_NONBLOCK' undeclared (first use in this function); did you mean 'SMTO_BLOCK'?
  return flags != -1 && fcntl(socket, F_SETFL, flags | O_NONBLOCK) != -1;
                                                       ^~~~~~~~~~
                                                       SMTO_BLOCK
kernel_libipsec_router.c: In function 'kernel_libipsec_router_create':
kernel_libipsec_router.c:338:6: warning: implicit declaration of function 'pipe'; did you mean '_pipe'? [-Wimplicit-function-declaration]
  if (pipe(this->notify) != 0 ||
      ^~~~
      _pipe
kernel_libipsec_router.c: In function 'set_nonblock':
kernel_libipsec_router.c:316:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
make[5]: *** [kernel_libipsec_router.lo] Error 1
make[5]: Leaving directory `/c/Download/strongswan-5.9.0/src/libcharon/plugins/kernel_libipsec'
make[4]: *** [all-recursive] Error 1
make[4]: Leaving directory `/c/Download/strongswan-5.9.0/src/libcharon'
make[3]: *** [all] Error 2
make[3]: Leaving directory `/c/Download/strongswan-5.9.0/src/libcharon'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/c/Download/strongswan-5.9.0/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/c/Download/strongswan-5.9.0'
make: *** [all] Error 2

Please help me to solve this problem, thank you very much!

**Comment by Thermi** [comment]
That won't work at all, ever, because it requires special code adaptations and a tun device driver. My employer and I work on getting such changes up and a prototype is working already.

**Reply by wyw900807** [reply]
Thank you for your reply.
